packages/iris-grid/src/sidebar/ChartBuilder.test.jsx

```diff
@@ -12,152 +13,114 @@ function makeChartBuilderWrapper({
+  chartButtons = container.querySelectorAll('.btn-chart-type');
 
-  wrapper.unmount();
+  expect(chartButtons[0].classList.contains('active')).toBe(false);
+  expect(chartButtons[1].classList.contains('active')).toBe(true);
+
+  unmount();
 });
 
 it('has x-axis selection with the proper columns', () => {
-  const wrapper = makeChartBuilderWrapper();
+  const { container, unmount } = makeChartBuilderWrapper();
+  const xAxisSelectElm = container.querySelector('.select-x-axis');
 
-  expect(wrapper.find('.select-x-axis').children().length).toBe(
+  expect(xAxisSelectElm.querySelectorAll('option').length).toBe(
     COLUMN_NAMES.length
   );
-  expect(wrapper.find('.select-x-axis').instance().value).toBe(COLUMN_NAMES[0]);
-  expect(wrapper.state('xAxis')).toBe(COLUMN_NAMES[0]);
-
-  wrapper.find('.select-x-axis').childAt(1).simulate('change');
-  expect(wrapper.state('xAxis')).toBe(COLUMN_NAMES[1]);
-
-  wrapper.unmount();
+  expect(xAxisSelectElm.value).toBe(COLUMN_NAMES[0]);
+  unmount();
 });
 
 it('updates series selection with the proper columns', () => {
-  const wrapper = makeChartBuilderWrapper();
+  const { container, unmount } = makeChartBuilderWrapper();
+  const seriesSelectElm = container.querySelector('.select-series');
 
-  expect(wrapper.find('.select-series').children().length).toBe(
+  expect(seriesSelectElm.querySelectorAll('option').length).toBe(
     COLUMN_NAMES.length
   );
-  expect(wrapper.find('.select-series').instance().value).toBe(COLUMN_NAMES[1]);
-  expect(wrapper.state('seriesItems').map(item => item.value)).toEqual([
-    COLUMN_NAMES[1],
-  ]);
+  expect(seriesSelectElm.value).toBe(COLUMN_NAMES[1]);
 
-  wrapper.find('.select-series').childAt(2).simulate('change');
-  expect(wrapper.state('seriesItems').map(item => item.value)).toEqual([
-    COLUMN_NAMES[2],
-  ]);
-
-  wrapper.unmount();
+  unmount();
 });
 
 it('add and deletes series items', () => {
-  const wrapper = makeChartBuilderWrapper();
-
-  expect(wrapper.state('seriesItems').length).toBe(1);
+  const { container } = makeChartBuilderWrapper();
+  const addSeriesItemBtn = container.querySelector('.btn-add-series');
 
-  wrapper.find('.btn-add-series').simulate('click');
-  wrapper.find('.btn-add-series').simulate('click');
+  expect(container.querySelectorAll('.form-series-item').length).toBe(1);
 
-  expect(wrapper.state('seriesItems').length).toBe(3);
+  userEvent.click(addSeriesItemBtn);
+  userEvent.click(addSeriesItemBtn);
 
-  wrapper.find('.select-series').at(1).childAt(3).simulate('change');
-  wrapper.find('.select-series').at(2).childAt(2).simulate('change');
-
-  expect(wrapper.state('seriesItems').map(item => item.value)).toEqual([
-    COLUMN_NAMES[1],
-    COLUMN_NAMES[3],
-    COLUMN_NAMES[2],
-  ]);
-
-  wrapper.find('.btn-delete-series').at(0).simulate('click');
-
-  expect(wrapper.state('seriesItems').map(item => item.value)).toEqual([
-    COLUMN_NAMES[3],
-    COLUMN_NAMES[2],
-  ]);
-});
-
-it('updates linked state', () => {
```

Comment:
I think we still need this check. Instead of looking at `isLinked` in the state, we can listen for `onChange` prop to get called with the new `isLinked` value.

Reply:
couldn't quite do it that way but I did add a check for the input's value